rfcs/for_in.md

```diff
@@ -0,0 +1,167 @@
+# `for...in`
+
+This RFC proposes to add a `for...in` construct in OCaml, similar to [Python's for-in](https://docs.python.org/3/reference/compound_stmts.html#for), [JavaScript's for-of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of), [Rust's for-in](https://doc.rust-lang.org/reference/expressions/loop-expr.html#iterator-loops), or [Lean's for-in](https://lean-lang.org/doc/reference/latest/Functors___-Monads-and--do--Notation/Syntax/#monad-iteration-syntax) (though that last one is more complex).
+
+
+## Overview
+
+The feature is straightforward, and would allow users to write something like[^1]:
+
+```ocaml
+for x in Iter.(1 -- 1000) do
+   Printf.printf "Number: %d" x
+done
+```
+
+The iterator must be of type `('a -> unit) -> unit`, corresponding to [popular implementations](https://ocaml.org/p/iter/1.6) of the iterator pattern in OCaml and to what the standard library offers -- modulo flipped arguments (see drawbacks).
```

Comment:
why this instead of Seq.t?

Reply:
`Seq.t` is less efficient in this context, as it requires unnecessary allocation of the intermediate closure.
`Seq.t` exists to allow inversion of control, which a for loop would not allow anyway.
Well, to be exact, inversion of control is now possible in both cases (including in for loops) using effect handlers, but the point stands